### PR TITLE
Bindu, Rohit | BAH-1475 | Upgrades openmrs version to 2.4.2 and mockito2 version to 2.0.7

### PR DESCRIPTION
**Pull request by binduak**
JIRA card -> https://bahmni.atlassian.net/browse/BAH-1475

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/rulesengine/engine/RulesEngineImpl.java [hunk 1]
```diff
@@ -21,7 +20,7 @@
 import java.util.List;
 import java.util.Set;
 
-@Component
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/rulesengine/engine/RulesEngineImpl.java:24]: there should be a corresponding bean def in the applicationContext, no?

File: api/src/main/java/org/openmrs/module/rulesengine/domain/OrderSetDrugRow.java [hunk 2]
```diff
@@ -2,9 +2,7 @@
 
 import org.bahmni.csv.CSVEntity;
 import org.bahmni.csv.annotation.CSVHeader;
-import org.springframework.stereotype.Component;
 
-@Component
```
Comment (angshu) [inline api/src/main/java/org/openmrs/module/rulesengine/domain/OrderSetDrugRow.java:7]: how is this being used? are we autowiring this from any place? or from other dependent modules? Should there be bean def in the application context xml

File: api/src/main/java/org/openmrs/module/rulesengine/engine/RulesEngineImpl.java (hunk 1 above)
Comment (binduak) [reply api/src/main/java/org/openmrs/module/rulesengine/engine/RulesEngineImpl.java:24]: @angshu throwing below exception 
_Annotation-specified bean name 'rulesEngine' for bean class conflicts with existing, non-compatible bean definition of same name and class._
as we are creating object in the below file.
https://github.com/Bahmni/openmrs-module-rulesengine/blob/master/api/src/main/java/org/openmrs/module/rulesengine/rule/CSVBasedDosageRule.java#L33

File: api/src/main/java/org/openmrs/module/rulesengine/domain/OrderSetDrugRow.java (hunk 2 above)
Comment (binduak) [reply api/src/main/java/org/openmrs/module/rulesengine/domain/OrderSetDrugRow.java:7]: @angshu we are not auto-wiring from rulesengine/bahmni-core repo's. So removed the unnecessary annotation.

**Pull request merged**